version: '3.8'

services:
  rekor:
    container_name: sigstore-rekor
    build:
      context: ./rekor
      dockerfile: Dockerfile
    ports:
    - "3000:3000" 
    networks:
      - sigstore-network
    restart: unless-stopped


  ctlog:
    container_name: sigstore-ctlog
    build:
      context: ./ctlog
      dockerfile: Dockerfile
    volumes:
      - fulcio-data:/shared-fulcio-config
      - ctlog-data:/shared-ctlog-config
    ports:
      - "6105:6105" 
    networks:
      - sigstore-network
    restart: unless-stopped

  fulcio:
    container_name: sigstore-fulcio
    build:
      context: ./fulcio
      dockerfile: Dockerfile
    volumes:
      - fulcio-data:/shared-fulcio-config
      - ${HOME}:/home
      - /sys/firmware/efi:/sys/firmware/efi
    ports:
      - "5000:5000" 
    networks:
      - sigstore-network
    user: root
    devices:
      - /dev/sgx_enclave
      - /dev/sgx_provision
    security_opt:
      - systempaths=unconfined
      - apparmor=unconfined
    restart: unless-stopped
    stdin_open: true
    tty: true


  securesystemlib:
    image: custom_gramine_ra_image
    container_name: securesystemlib
    user: root
    devices:
      - /dev/sgx_enclave
      - /dev/sgx_provision
    security_opt:
      - systempaths=unconfined
      - apparmor=unconfined
    volumes:
      - ${HOME}:/home
      - /sys/firmware/efi:/sys/firmware/efi
    networks:
      - sigstore-network
    stdin_open: true
    tty: true
    restart: unless-stopped

volumes:
  fulcio-data:
    name: fulcio-data
  ctlog-data:
    name: ctlog-data

networks:
  sigstore-network:
    driver: bridge
    name: sigstore-network

    